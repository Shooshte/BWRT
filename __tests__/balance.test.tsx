import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import Balance from "../src/app/components/topBar/balance";
import { Wrapper } from "../src/tests/test-utils";

describe("Balance", () => {
  test("Render state balance", async () => {
    render(<Balance />, { wrapper: Wrapper });
    const BTCBalance = await screen.getByText(
      (_, element) => element?.textContent === "0.11342278 BTC"
    );
    expect(BTCBalance).toBeInTheDocument();
    const USDBalance = await screen.getByText(
      (_, element) => element?.textContent === "225.05 $"
    );
    expect(USDBalance).toBeInTheDocument();
    const AvailableText = await screen.getByText("Available");
    expect(AvailableText).toBeInTheDocument();
  });

  test("Render correct dom structure", () => {
    const { container } = render(
      <div style={{ height: "100%", width: "100%" }}>
        <Balance />
      </div>,
      { wrapper: Wrapper }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
