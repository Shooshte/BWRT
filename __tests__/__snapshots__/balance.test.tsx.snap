// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance Render correct dom structure 1`] = `
<div
  style="height: 100%; width: 100%;"
>
  <div
    class="container undefined"
  >
    <p>
      Available
    </p>
    <p>
      0.11342278
       
      <b>
        BTC
      </b>
    </p>
    <p>
      225.05
       
      <b>
        $
      </b>
    </p>
  </div>
</div>
`;
